plugins/jobs/src/js/JobsRunNow.js

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable no-unused-vars */
+import React from "react";
+/* eslint-enable no-unused-vars */
+
+import { graphqlObservable } from "data-service";
+import gql from "graphql-tag";
+import "rxjs/add/operator/concatMap";
+
+import { Subject } from "rxjs/Subject";
+import { defaultSchema } from "./JobsMenuClient";
+
+const runNowJobMutation = gql`
+  mutation {
+    runJob(id: $id) {
+      id
+    }
+  }
+`;
+
+const runNowJobGraphql = id => {
+  return graphqlObservable(runNowJobMutation, defaultSchema, id);
+};
+
+export const runNowEvent$ = new Subject();
+export const jobsRunNow$ = runNowEvent$.switchMap(id => {
```

Comment:
@DanielMSchmidt Over here we use switchmap with the runNowEvent Subject to actually listen to this observable but also have the ability to trigger it as well.